common/lib/protocol-definitions/src/protocol.ts

```diff
@@ -207,13 +207,26 @@ export interface ISummaryContent {
 
 /**
  * General errors returned from the server.
- * May want to add error code or something similar in the future.
  */
 export interface IServerError {
     /**
-     * Message describing the server error.
+     * An error code number that represents the error. It will be a valid HTTP error code.
+     * 403 errors are non retryable.
+     * 400 errors are always immediately retriable.
+     * 429 errors are retriable or non retriable (depends on type field).
      */
-    errorMessage: string;
+    code: number;
```

Comment:
this is technically breaking, but probably ok since it's unused except to extend in this file, but as i mention below i don't think the extending is worth it. with duck typing there isn't much value have a shared type, as as long as they have matching members you are good.